### PR TITLE
Add start datetime parameter to gsp/pvlive endpoint

### PR DESCRIPTION
**Pull request by devsjc**
# Adds the start_datetime_utc parameter to gsp/pvlive to enable specification of a start datetime

### CODE REVIEW

**Comment by devsjc** [comment]
@braddf @rachel-labri-tipton Only thing I'm not sure about here is, if we're setting the usual number of days back that this endpoing can access behind an environment variable on the server side, perhaps there was the intention to keep whatever we set as N_HISTORY_DAYS as the maximum that a client-side user could ask for? This would overwrite that - so maybe I should put a block in to limit it? Or remove start datetime as an option and just change it to something like "snapshot" as a bool which just sets the startdate to last 15 minutes, preventing abuse?

File: src/gsp.py [hunk 1]
```diff
@@ -228,6 +230,8 @@ def get_truths_for_a_specific_gsp(
     #### Parameters
     - **gsp_id**: _gsp_id_ of the requested forecast
     - **regime**: can choose __in-day__ or __day-after__
+    - **start_datetime_utc**: optional start datetime for the query.
+    If not set, defaults to N_HISTORY_DAYS env var, which if not set defaults to yesterday.
```
Comment (rachel-labri-tipton) [inline src/gsp.py:234]: It could be helpful to say what format the `start_datetime_utc` should be in. ex. 2023-07-22...

**Comment by devsjc** [comment]
> @devsjc, I went ahead and ran the tests on this and the route looks good.
> I could be missing something, but how is having a "snapshot" (which is better named that "historic") different from having the "historic" bool?

It might not be much better! It could just be better than target_datetime as an option as it would prevent a user trying to access the entire contents of the table with one call via a very early start_datetime_utc.

File: src/gsp.py (hunk 1 above)
Comment (peterdudfield) [reply src/gsp.py:234]: I think its a `datetime` so you dont have to specific the formant

File: src/tests/conftest.py [hunk 2]
```diff
@@ -25,12 +25,18 @@ def forecasts(db_session):
 @pytest.fixture
 def db_connection():
     """Pytest fixture for a database connection"""
-    # with tempfile.NamedTemporaryFile(suffix="db") as tmp:
-    #     set url option to not check same thread, this solves an error seen in testing
 
-    # url = f"sqlite:///{tmp.name}.db?check_same_thread=False"
-    # os.environ["DB_URL"] = url
-    # os.environ["DB_URL_PV"] = url
+    # -- Uncomment for dockerised testing --
```
Comment (peterdudfield) [inline src/tests/conftest.py:29]: should we use this all the time?

Comment (devsjc) [reply src/tests/conftest.py:29]: Well I reckon so, however it didn't work in github actions when I tried it and didn't think it was worth the time investment to figure it out just yet. Odd thing is, the other method doesn't work locally for me...!

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
> > @devsjc, I went ahead and ran the tests on this and the route looks good.
> > I could be missing something, but how is having a "snapshot" (which is better named that "historic") different from having the "historic" bool?
> 
> It might not be much better! It could just be better than target_datetime as an option as it would prevent a user trying to access the entire contents of the table with one call via a very early start_datetime_utc.

I like the idea of this, I would perhaps put the limit on, so that the user can get the whole history. Thats what PVlive is for. Could you do this @devsjc ?

**Review by peterdudfield — CHANGES_REQUESTED** [review]
add limit, so can only pull at max 3 days of data

**Comment by peterdudfield** [comment]
This helps with #269

**Pull request merged**